### PR TITLE
machines: Make ruff happy again

### PR DESCRIPTION
**Pull request by mvollmer**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [tool.ruff]
-select = [
+lint.select = [
```
Comment (allisonkarlitskaya) [inline pyproject.toml:2]: can you rather move these to a new `[tool.ruff.lint]` section?

File: test/run [hunk 2]
```diff
@@ -19,9 +19,9 @@ test_mypy() {
 }
 
 test_ruff
-test_mypy
+#test_mypy
```
Comment (allisonkarlitskaya) [inline test/run:22]: Is this for testing?  (and x2 more below)

File: machine/machine_core/ssh_connection.py [hunk 3]
```diff
@@ -32,7 +32,7 @@
 # HACK: some projects directly import ssh_connection before adjusting sys.path; add bots root dir
 sys.path.insert(1, os.path.dirname(os.path.dirname(os.path.dirname(os.path.realpath(__file__)))))
 
-from lib.constants import TEST_DIR  # noqa: E402
+from lib.constants import TEST_DIR
```
Comment (allisonkarlitskaya) [inline machine/machine_core/ssh_connection.py:35]: According to my understanding, this noqa should still beq required.  Or did ruff (very) recently get smarter about this?

**Review by allisonkarlitskaya — CHANGES_REQUESTED** [review]

File: test/run (hunk 2 above)
Comment (mvollmer) [reply test/run:22]: Ouch, yes.

File: pyproject.toml (hunk 1 above)
Comment (mvollmer) [reply pyproject.toml:2]: Done.

**Comment by mvollmer** [comment]
I can't access the "unit-tests" results, the URL is `https://localhost.localdomain:9000/logs/pull-5873-20240205-124243-664ff424-unit-tests/log.html`.

**Comment by allisonkarlitskaya** [comment]
> I can't access the "unit-tests" results, the URL is `https://localhost.localdomain:9000/logs/pull-5873-20240205-124243-664ff424-unit-tests/log.html`.

The `cockpituous` workflow does a test of CI by running a tasks container locally and having it process this PR as if it were any other.  The `unit-tests` status that results from that is pointing back to the inside of that container.

**Review by allisonkarlitskaya — APPROVED** [review]
Fixed with an update to the tasks container — it needed the new version of ruff in order to agree with your changes.  Let's land this.

**Pull request merged**